src/model/selection.js

```diff
@@ -829,7 +829,9 @@ function isUnvisitedBlock( element, visited ) {
 
 	visited.add( element );
 
-	return element.document.model.schema.isBlock( element ) && element.parent;
+	const document = element.is( 'rootElement' ) ? element.document : element.root.document;
```

Comment:
That's not necessary. `element.root.document` will always work, even if `element` is a root.